src/providers/details/tabs/Credentials/components/CredentialsSectionEditViewModes.tsx

```diff
@@ -0,0 +1,32 @@
+import { type FC, useState } from 'react';
+
+import type { IoK8sApiCoreV1Secret, V1beta1Provider } from '@kubev2v/types';
+
+import CredentialsSectionEditMode from './CredentialsSectionEditMode';
+import CredentialsSectionViewMode from './CredentialsSectionViewMode';
+
+import './CredentialsSectionEditViewModes.style.css';
+
+type CredentialsSectionEditViewModesProps = {
+  provider: V1beta1Provider;
+  secret: IoK8sApiCoreV1Secret;
+};
+
+const CredentialsSectionEditViewModes: FC<CredentialsSectionEditViewModesProps> = ({
+  provider,
+  secret,
+}) => {
+  const [editMode, setEditMode] = useState(false);
+
+  const toggleEdit = () => {
+    setEditMode(!editMode);
```

Comment:
It's always better to use the arg value than the external value - setEditMode(isEdit => !isEdit);